This was the panic stack trace prior to this fix:
```
❯ RCLONE_CONFIG=rclone.conf.txt rclone ls mega:
Enter configuration password:
password:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55ed848c3ae7]

goroutine 1 gp=0x148580fc21e0 m=9 mp=0x148581493008 [running]:
panic({0x55ed88b15680?, 0x55ed891dd550?})
        /usr/lib/golang/src/runtime/panic.go:879 +0x16f fp=0x14858102f3f0 sp=0x14858102f340 pc=0x55ed8366ce8f
runtime.panicmem(...)
        /usr/lib/golang/src/runtime/panic.go:336
runtime.sigpanic()
        /usr/lib/golang/src/runtime/signal_unix.go:931 +0x378 fp=0x14858102f450 sp=0x14858102f3f0 pc=0x55ed836705b8
github.com/t3rm1n4l/go-mega.(*Node).GetType(0x148581448840?)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/vendor/github.com/t3rm1n4l/go-mega/mega.go:188 +0x27 fp=0x14858102f498 sp=0x14858102f450 pc=0x55ed848c3ae7
github.com/rclone/rclone/backend/mega.(*Fs).findDir(0x0?, 0x0?, {0x7ffe8c4059f7?, 0x55ed848c8dc0?})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/backend/mega/mega.go:344 +0x6a fp=0x14858102f4e0 sp=0x14858102f498 pc=0x55ed848d5f8a
github.com/rclone/rclone/backend/mega.(*Fs).findRoot(0x148581448840, {0x55ed891008c0, 0x55ed895df100}, 0x0)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/backend/mega/mega.go:449 +0x108 fp=0x14858102f570 sp=0x14858102f4e0 pc=0x55ed848d6ec8
github.com/rclone/rclone/backend/mega.NewFs({0x55ed891008c0, 0x55ed895df100}, {0x7ffe8c4059f7, 0x4}, {0x7ffe8c4059f7, 0x0}, {0x55ed890e7ec8, 0x1485816fc600})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/backend/mega/mega.go:294 +0x9aa fp=0x14858102f6c0 sp=0x14858102f570 pc=0x55ed848d5aaa
github.com/rclone/rclone/fs.NewFs({0x55ed891008c0, 0x55ed895df100}, {0x7ffe8c4059f7, 0x5})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/fs/newfs.go:75 +0x370 fp=0x14858102f820 sp=0x14858102f6c0 pc=0x55ed83a2b130
github.com/rclone/rclone/fs/cache.GetFn.func1({0x7ffe8c4059f7?, 0x155ed83846fb4?})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/fs/cache/cache.go:103 +0x38 fp=0x14858102f878 sp=0x14858102f820 pc=0x55ed83a54ed8
github.com/rclone/rclone/lib/cache.(*Cache).Get(0x1485816f71a0, {0x7ffe8c4059f7, 0x5}, 0x14858102f9e0)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/lib/cache/cache.go:88 +0xe6 fp=0x14858102f8f0 sp=0x14858102f878 pc=0x55ed83a52ba6
github.com/rclone/rclone/fs/cache.GetFn({0x55ed891008c0, 0x55ed895df100}, {0x7ffe8c4059f7, 0x5}, 0x55ed890b9498)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/fs/cache/cache.go:102 +0x105 fp=0x14858102fa28 sp=0x14858102f8f0 pc=0x55ed83a54865
github.com/rclone/rclone/fs/cache.Get({0x55ed891008c0, 0x55ed895df100}, {0x7ffe8c4059f7, 0x5})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/fs/cache/cache.go:183 +0x85 fp=0x14858102fa80 sp=0x14858102fa28 pc=0x55ed83a552c5
github.com/rclone/rclone/cmd.NewFsFile({0x7ffe8c4059f7, 0x5})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/cmd/cmd.go:92 +0x113 fp=0x14858102fb18 sp=0x14858102fa80 pc=0x55ed855573f3
github.com/rclone/rclone/cmd.newFsFileAddFilter({0x7ffe8c4059f7, 0x5})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/cmd/cmd.go:114 +0x57 fp=0x14858102fbc8 sp=0x14858102fb18 pc=0x55ed85557677
github.com/rclone/rclone/cmd.NewFsSrc(...)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/cmd/cmd.go:136
github.com/rclone/rclone/cmd/ls.init.func1(0x55ed89253d40, {0x148581125370, 0x1, 0x55ed869ba7f1?})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/cmd/ls/ls.go:40 +0x56 fp=0x14858102fc20 sp=0x14858102fbc8 pc=0x55ed856d6c36
github.com/spf13/cobra.(*Command).execute(0x55ed89253d40, {0x148581125330, 0x1, 0x1})
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/vendor/github.com/spf13/cobra/command.go:1019 +0xb07 fp=0x14858102fdc8 sp=0x14858102fc20 pc=0x55ed85546f07
github.com/spf13/cobra.(*Command).ExecuteC(0x55ed89249680)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/vendor/github.com/spf13/cobra/command.go:1148 +0x465 fp=0x14858102feb8 sp=0x14858102fdc8 pc=0x55ed855478a5
github.com/spf13/cobra.(*Command).Execute(...)
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/vendor/github.com/spf13/cobra/command.go:1071
github.com/rclone/rclone/cmd.Main()
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/cmd/cmd.go:541 +0x31 fp=0x14858102ff38 sp=0x14858102feb8 pc=0x55ed8555a311
main.main()
        /builddir/build/BUILD/rclone-1.73.0-build/rclone-1.73.0/rclone.go:14 +0xf fp=0x14858102ff48 sp=0x14858102ff38 pc=0x55ed858d112f
runtime.main()
        /usr/lib/golang/src/runtime/proc.go:290 +0x2d5 fp=0x14858102ffe0 sp=0x14858102ff48 pc=0x55ed83634435
runtime.goexit({})
        /usr/lib/golang/src/runtime/asm_amd64.s:1771 +0x1 fp=0x14858102ffe8 sp=0x14858102ffe0 pc=0x55ed836764a1
```

and this is what the failure looks like with this PR + https://github.com/t3rm1n4l/go-mega/pull/61:

```
❯ RCLONE_CONFIG=rclone.conf.txt ~/go/bin/rclone ls mega:
Enter configuration password:
password:
2026/04/12 17:27:18 CRITICAL: Failed to create file system for "mega:": login with previous auth keys failed: Post "https://g.api.mega.co.nz/cs?id=<redacted>&sid=<redacted>": dial tcp [2a0b:e46:1:100::15]:443: connect: software caused connection abort
```

--- a/backend/mega/mega.go
+++ b/backend/mega/mega.go
@@ -284,7 +284,7 @@ func NewFs(ctx context.Context, name, root string, m configmap.Mapper) (fs.Fs, e
 			}
 			err = srv.LoginWithKeys(opt.SessionID, decodedMasterKey)
 			if err != nil {
-				fs.Debugf(f, "login with previous auth keys failed: %v", err)
+				return nil, fmt.Errorf("login with previous auth keys failed: %w", err)
 			}
 		}
 	}

